spec/unit/shoulda/matchers/active_model/validate_numericality_of_matcher_spec.rb

```diff
@@ -1157,7 +1247,15 @@ def validate_numericality
     validate_numericality_of(attribute_name)
   end
 
+  def validate_virtual_attribute_numericality
+    validate_numericality_of(virtual_attribute_name)
+  end
+
   def attribute_name
     :attr
   end
+
+  def virtual_attribute_name
```

Comment:
I think we can simply reuse `attribute_name` within `define_model_validating_numericality_of_virtual_attribute`. I suppose it's helpful conceptually, but it's not like we use both `attribute_name` and `virtual_attribute_name` in the same test. This means we could remove `validate_virtual_attribute_numericality`.